File: client/src/types/post.ts
import type { PostCategoryResponse } from "./post-category";
import type { PostTagResponse } from "./post-tag";
import type { SeriesResponse, PostLink } from "./series";

export interface PostResponse {
  id: string;
  slug: string;
  title: string;
  description?: string;
  content: string;
  coverImage?: string;
  createdAt: string;
  updatedAt: string;
  published: boolean;
  featured: boolean;
  category: PostCategoryResponse;
  tags: PostTagResponse[];
  series?: SeriesResponse;
  seriesOrder?: number;
  prevPost?: PostLink;
  nextPost?: PostLink;
  readingTime: number;
  wordCount: number;
  views: number;
}

export interface PostArchiveResponse {
  id: string;
  slug: string;
  title: string;
  description?: string;
  coverImage?: string;
  createdAt: string;
  updatedAt: string;
  published: boolean;
  featured: boolean;
  category: PostCategoryResponse;
  tags: PostTagResponse[];
  readingTime: number;
  wordCount: number;
  views: number;
}

export interface ArchiveItem {
  year: number;
  month: number;
  posts: PostArchiveResponse[];
}

export interface PostCreate {
  title: string;
  slug: string;
  description?: string;
  content: string;
  coverImage?: string;
  published?: boolean;
  featured?: boolean;
  categoryId: string;
  tagIds: string[];
  seriesId?: string;
  seriesOrder?: number;
}

export interface PostUpdate {
  title?: string;
  slug?: string;
  description?: string;
  content?: string;
  coverImage?: string;
  published?: boolean;
  featured?: boolean;
  categoryId?: string;
  tagIds?: string[];
  seriesId?: string;
  seriesOrder?: number;
}

File: server/app/schemas/post.py
from pydantic import BaseModel, Field

from app.schemas.post_category import PostCategoryResponse
from app.schemas.post_tag import PostTagResponse

# ---------- 创建 / 更新 ----------
class PostCreate(BaseModel):
    slug: str
    title: str
    description: str | None = None
    content: str
    cover_image: str | None = Field(None, alias="coverImage")
    published: bool = False
    featured: bool = False
    category_id: str = Field(..., alias="categoryId")
    tag_ids: list[str] = Field(default_factory=list, alias="tagIds")
    reading_time: int = Field(default=0, alias="readingTime")
    word_count: int = Field(default=0, alias="wordCount")

    model_config = {"populate_by_name": True}


class PostUpdate(BaseModel):
    slug: str | None = None
    title: str | None = None
    description: str | None = None
    content: str | None = None
    cover_image: str | None = Field(None, alias="coverImage")
    published: bool | None = None
    featured: bool | None = None
    category_id: str = Field(..., alias="categoryId")
    tag_ids: list[str] = Field(default_factory=list, alias="tagIds")
    reading_time: int = Field(default=0, alias="readingTime")
    word_count: int = Field(default=0, alias="wordCount")

    model_config = {"populate_by_name": True}

# ---------- 响应 ----------


class PostResponse(BaseModel):
    id: str
    slug: str
    title: str
    description: str | None = None
    content: str
    cover_image: str | None = Field(None, alias="coverImage")
    created_at: str = Field(..., alias="createdAt")
    updated_at: str = Field(..., alias="updatedAt")
    published: bool
    category: PostCategoryResponse | None = None
    tags: list[PostTagResponse] = Field(default_factory=list)
    reading_time: int = Field(default=0, alias="readingTime")
    word_count: int = Field(default=0, alias="wordCount")
    views: int
    featured: bool

    model_config = {"from_attributes": True, "populate_by_name": True}


# ---------- 列表 ----------
class PostListItem(BaseModel):
    id: str
    slug: str
    title: str
    description: str | None = None
    content: str
    cover_image: str | None = Field(None, alias="coverImage")
    created_at: str = Field(..., alias="createdAt")
    updated_at: str = Field(..., alias="updatedAt")
    published: bool
    featured: bool
    category: PostCategoryResponse | None = None
    tags: list[PostTagResponse] = Field(default_factory=list)
    reading_time: int = Field(default=0, alias="readingTime")
    word_count: int = Field(default=0, alias="wordCount")
    views: int

    model_config = {"from_attributes": True, "populate_by_name": True}


# ---------- 归档 ----------
class ArchiveItem(BaseModel):
    year: int
    month: int
    posts: list[PostListItem]
